torchaudio/functional.py

```diff
@@ -844,6 +844,152 @@ def compute_deltas(specgram, win_length=5, mode="replicate"):
+
+    Args:
+       waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+       gain_db (float) Gain adjustment in decibels (dB) (Default: `1.0`).
+
+    Returns:
+       torch.Tensor: waveform amplified by gain_db
+    """
+    if (gain_db == 0):
+        return waveform
+
+    ratio = 10 ** (gain_db / 20)
+
+    return waveform * ratio
+
+
+@torch.jit.script
+def scale_to_interval(waveform, interval=1.0):
+    # type: (Tensor, float) -> Tensor
+    r"""Scales waveform to an interval.
+    Args:
+       waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+       interval (float): The bounds of the interval, where the float indicates
+           the upper bound and the negative of the float indicates the lower
+           bound (Default: `1.0`)
+           Example: interval=1.0 -> [-1.0, 1.0]
+    Returns:
+       torch.Tensor: waveform scaled to interval
+    """
+    abs_max = torch.max(torch.abs(waveform))
+    ratio = abs_max / interval
+    waveform /= ratio
+
+    return waveform
+
+
+def _add_noise_shaping(dithered_waveform, waveform):
+    r"""Noise shaping is calculated by error:
+    error[n] = dithered[n] - original[n]
+    noise_shaped_waveform[n] = dithered[n] + error[n-1]
+    """
+    error = dithered_waveform - waveform
+
+    # add error[n-1] to dithered_waveform[n], so offset the error by 1 index
+    for index in range(error.size()[0]):
+        err = error[index]
+        error_offset = torch.cat((torch.zeros(1), err))
+        error[index] = error_offset[:waveform.size()[1]]
+
+    noise_shaped = dithered_waveform + error
+    return noise_shaped
+
+
+@torch.jit.script
+def probability_distribution(waveform, density_function="TPDF"):
+    # type: (Tensor, str) -> Tensor
+    r"""Apply a probability distribution function on a waveform.
+
+    Triangular probability density function (TPDF) dither noise has a 
+    triangular distribution; values in the center of the range have a higher 
+    probability of occurring. 
+
+    Rectangular probability density function (RPDF) dither noise has a 
+    uniform distribution; any value in the specified range has the same
+    probability of occurring.
+
+    Gaussian probability density function (GPDF) has a normal distribution.
+    The relationship of probabilities of results follows a bell-shaped,
+    or Gaussian curve, typical of dither generated by analog sources.
+    Args:
+        waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+        probability_density_function (string): The density function of a
+           continuous random variable (Default: `TPDF`)
+           Options: Triangular Probability Density Function - `TPDF`
+                    Rectangular Probability Density Function - `RPDF`
+                    Gaussian Probability Density Function - `GPDF`
+    Returns:
+        torch.Tensor: waveform dithered with TPDF
+    """
+    wave_size = waveform.size()[1] - 1
```

Comment:
This assumes a waveform of the form (channel, time), right? It would be more easily batchable to assume (..., time), and put -1 instead of 1 here.

There's a few more of these manipulation that trickle in here. If this is too much effort, we could make it a separate PR to make the transforms batchable.